plugins/woocommerce/includes/wc-conditional-functions.php

```diff
@@ -127,12 +127,14 @@ function is_checkout() {
 	/**
 	 * Is_checkout_pay - Returns true when viewing the checkout's pay page.
 	 *
+	 * @param bool $check_key Optional. If true, check if the key is set in the URL.
 	 * @return bool
 	 */
-	function is_checkout_pay_page() {
+	function is_checkout_pay_page( bool $check_key = false ): bool {
 		global $wp;
 
-		return is_checkout() && ! empty( $wp->query_vars['order-pay'] );
+		$is_pay_for_order = ( is_checkout() && ! empty( $wp->query_vars['order-pay'] ) ) || is_wc_endpoint_url( 'order-pay' ) || isset( $_GET['pay_for_order'] );  // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

Comment:
We need a thorough comment on the new conditions:
- `is_wc_endpoint_url( 'order-pay' )`: use case and clues on the bigger picture
- `isset( $_GET['pay_for_order'] )`: same
- `( ! $check_key || isset( $_GET['key'] ) )`: same

Inspite I dislike `if (...) { return ...; } ` constructs, here it can be good application.
Otherwise, understanding why or reacting to this function further is hard (at least for me).